build/BuildSteps.cs

```diff
@@ -643,10 +642,22 @@ public static void DotnetPublishForNupkg()
             Shell.Run("dotnet", $"publish {Settings.ProjectFile} " +
                                 $"/p:BuildNumber=\"{Settings.BuildNumber}\" " +
                                 $"/p:NoWorkers=\"true\" " +
-                                $"/p:TargetFramework=net6.0 " +
+                                $"/p:TargetFramework=net8.0 " +
+                                $"/p:CommitHash=\"{Settings.CommitId}\" " +
+                                (string.IsNullOrEmpty(Settings.IntegrationBuildNumber) ? string.Empty : $"/p:IntegrationBuildNumber=\"{Settings.IntegrationBuildNumber}\" ") +
+                                $"-c Release -f net8.0");
+        }
+
+        public static void DotnetPublishForCustomHost()
+        {
+            // This publishes to the /bin/Release/$targetFramework$/publish
+            Shell.Run("dotnet", $"publish {Settings.CustomHostProjectFile} " +
+                                $"/p:BuildNumber=\"{Settings.BuildNumber}\" " +
+                                $"/p:NoWorkers=\"true\" " +
+                                $"/p:TargetFramework=net8.0 " +
                                 $"/p:CommitHash=\"{Settings.CommitId}\" " +
                                 (string.IsNullOrEmpty(Settings.IntegrationBuildNumber) ? string.Empty : $"/p:IntegrationBuildNumber=\"{Settings.IntegrationBuildNumber}\" ") +
-                                $"-c Release -f net6.0");
+                                $"-c Release -f net8.0 -o {Settings.NupkgPublishDir}");
```

Comment:
Where will the output of this goes? Don't we need this in the root of the package? We need to make sure we are AOT publishing on the correct platform (not just windows).

Reply:
so when I was testing locally without specifying the output directory it would go to `host/src/FunctionsCustomHost/bin/Release/net8.0/{framework}/publish`. With the specified output directory it goes to `/bin/Release/{targetFramework}/publish` which is what I thought we wanted

Reply:
removed